sql-db/vertx-sql/src/test/java/io/quarkus/qe/vertx/sql/dbpool/PostgresPoolIT.java

```diff
@@ -127,27 +120,45 @@ public void checkIdleExpirationTime() throws InterruptedException {
         done.await(TIMEOUT_SEC, TimeUnit.SECONDS);
         assertEquals(done.getCount(), 0, String.format("Missing %d events.", events - done.getCount()));
 
-        Long connectionsAmount = addAnotherConection();
-        assertEquals(1, connectionsAmount, "Idle doesn't remove IDLE expired connections!.");
+        // Make just one extra query and Hold "Idle + 1 sec" in order to release inactive connections.
+        Uni<Long> activeConnectionAmount = selectActiveConnectionsAfterConnection();
+
+        CountDownLatch doneIdleExpired = new CountDownLatch(1);
+        activeConnectionAmount.subscribe().with(connectionsAmount -> {
+            // At this point you should just have one connection -> SELECT CURRENT_TIMESTAMP
+            assertEquals(1, connectionsAmount, "Idle doesn't remove IDLE expired connections!.");
+            if (connectionsAmount == 1) {
+                doneIdleExpired.countDown();
+            }
+        });
+
+        doneIdleExpired.await(TIMEOUT_SEC, TimeUnit.SECONDS);
+        assertEquals(doneIdleExpired.getCount(), 0, "Missing doneIdleExpired query.");
+    }
+
+    private Uni<Long> activeConnections() {
```

Comment:
@fedinskiy fyi, this is the change that is making the test worked in JVM and Native both.